### PR TITLE
Update versions of GH actions

### PR DESCRIPTION
**Pull request by krassowski**
Gets rid of Node 12 deprecation warnings

### CODE REVIEW

**Comment by krassowski** [comment]
This PR and [scheduled checks](https://github.com/jupyterlab/extension-cookiecutter-ts/actions/runs/3666531162) are failing on Python 3.11 with: `ImportError: Error importing plugin "pytest_jupyter.jupyter_server": No module named 'pytest_jupyter'`. I don't think pytest_jupyter is an explicit dependency in cookiecutter so I am a bit lost. @blink1073 is this something that needs to be fixed upstream in maintainer-tools or similar?

**Comment by fcollonval** [comment]
We need to explicitly use it by adding the dependency there

https://github.com/jupyterlab/extension-cookiecutter-ts/blob/4fca73f34a69bb6ce1e88b4f7eef6d56d92c0c3d/%7B%7Bcookiecutter.python_name%7D%7D/pyproject.toml#L31

and switching the configuration there

https://github.com/jupyterlab/extension-cookiecutter-ts/blob/4fca73f34a69bb6ce1e88b4f7eef6d56d92c0c3d/%7B%7Bcookiecutter.python_name%7D%7D/conftest.py#L3

The later command is the reason of the error because it ends up loading that configuration file requiring `pytest_jupyter`:

https://github.com/jupyter-server/jupyter_server/blob/main/jupyter_server/pytest_plugin.py

File: {{cookiecutter.python_name}}/pyproject.toml [hunk 1]
```diff
@@ -33,6 +33,7 @@ test = [{% if cookiecutter.test.lower().startswith('y') %}
     "pytest",
     "pytest-asyncio",
```
Comment (blink1073) [inline {{cookiecutter.python_name}}/pyproject.toml:34]: You shouldn't need pytest-asyncio or pytest-tornasync if you're using pytest-jupyter.

Comment (blink1073) [reply {{cookiecutter.python_name}}/pyproject.toml:34]: Actually, you'd need pytest-asyncio in order to write async fixtures.

Comment (fcollonval) [reply {{cookiecutter.python_name}}/pyproject.toml:34]: Thanks for the review @blink1073 

I removed `pytest-tornasync`

**Review by blink1073 — APPROVED** [review]
LGTM, thanks!

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.0

**Pull request merged**